### PR TITLE
Fix edge case in traversal and add more test cases for invalid inputs

### PR DESCRIPTION
**Pull request by angary**
- Change to use `strconv.Atoi` instead of `strconv.ParseInt`
- Add bonus test cases for invalid case
- Return error if the ascii to integer fails (if there was an error, index would be 0) resulting in invalid field)

Also wondering if there is a way to do deep equality checks - this would make the traversal testing less rigid, i.e. we check that the struct returned from traversal has deep equality with struct in `setupDocument`

### CODE REVIEW

**Comment by Varun-Sethu** [comment]
To answer your question, IIRC there is a `reflect.DeepEqual`

File: Backend/editor/models/tests/models_test.go [hunk 1]
```diff
@@ -79,7 +66,19 @@ func TestSliceField(t *testing.T) {
...
 
-func TestArrayField(t *testing.T) {
+func TestInvalidFieldName(t *testing.T) {
+	testObj := setupDocument()
+	path := "Content/0/InvalidFieldName/0"
+	subpaths, err := models.ParsePath(path)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	result, err := models.Traverse(testObj, subpaths)
```
Comment (Varun-Sethu) [inline Backend/editor/models/tests/models_test.go:76]:
```go
_, err := models.Traverse(testObj, subpaths)
```

File: Backend/editor/models/tests/models_test.go [hunk 2]
```diff
@@ -95,7 +94,31 @@ func TestArrayField(t *testing.T) {
...
 
-func TestStructField(t *testing.T) {
+func TestNonIntegerArrayIndex(t *testing.T) {
+	testObj := setupDocument()
+	path := "Content/asdf/Data"
+	subpaths, err := models.ParsePath(path)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	result, err := models.Traverse(testObj, subpaths)
```
Comment (Varun-Sethu) [inline Backend/editor/models/tests/models_test.go:104]: Same here

File: Backend/editor/models/tests/models_test.go [hunk 3]
```diff
@@ -95,7 +94,31 @@ func TestArrayField(t *testing.T) {
...
+}
+
+func TestOutOfBoundsArrayIndex(t *testing.T) {
+	testObj := setupDocument()
+	path := "Content/4/Data"
+	subpaths, err := models.ParsePath(path)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	result, err := models.Traverse(testObj, subpaths)
```
Comment (Varun-Sethu) [inline Backend/editor/models/tests/models_test.go:116]: and here

**Review by Varun-Sethu — APPROVED** [review]

**Review by brianzhang2101 — APPROVED** [review]
lgtm
